### PR TITLE
Fix Jira Server Plugin Installation Validation

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

File: src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java [hunk 1]
```diff
@@ -95,8 +96,6 @@ public JiraCustomEndpoint(JiraChannelKey jiraChannelKey, CustomEndpointManager c
             Thread.currentThread().interrupt();
             return Optional.of(responseFactory.createInternalServerErrorResponse("", "Thread was interrupted while validating Jira plugin installation: " + e.getMessage()));
         }
-
-        return Optional.empty();
```
Comment (gkillough) [inline src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java:99]: Having the `return Optional.empty()` after the try-catch makes it easy to forget to return/throw errors.

File: src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomEndpoint.java [hunk 2]
```diff
@@ -92,15 +92,14 @@ public JiraServerCustomEndpoint(JiraServerChannelKey jiraChannelKey, CustomEndpo
             if (!jiraPluginInstalled) {
                 return Optional.of(responseFactory.createNotFoundResponse("Was not able to confirm Jira server successfully installed the Jira Server plugin. Please verify the installation on you Jira server."));
             }
+            return Optional.empty();
         } catch (IntegrationException e) {
-            createBadRequestIntegrationException(e);
+            return createBadRequestIntegrationException(e);
```
Comment (gkillough) [inline src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomEndpoint.java:97]: The actual fix.

File: src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java [hunk 3]
```diff
@@ -87,6 +87,7 @@ public JiraCustomEndpoint(JiraChannelKey jiraChannelKey, CustomEndpointManager c
             if (!jiraPluginInstalled) {
                 return Optional.of(responseFactory.createNotFoundResponse("Was not able to confirm Jira Cloud successfully installed the Jira Cloud plugin. Please verify the installation on you Jira Cloud server."));
             }
+            return Optional.empty();
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java:90]: Is this change so we are consistent with the jira server change?

File: src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomEndpoint.java (hunk 2 above)
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomEndpoint.java:97]: Was this the cause of the issue?

File: src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java (hunk 3 above)
Comment (gkillough) [reply src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java:90]: Yes. Just to ensure that we don't forget to actually return the error message.

**Pull request merged**